torchgen/gen.py

```diff
@@ -1955,16 +1980,36 @@ def gen_per_operator_headers(
         ("NativeMetaFunctions", "_meta"),
         ("NativeFunctions", "_native"),
     ]:
-        cpu_fm.write(
-            f"{category}.h",
-            lambda: {
-                f"{category}_includes": [
-                    f"#include <ATen/ops/{name}{suffix}.h>"
-                    for name in sorted(functions_by_root_name.keys())
-                ],
-                f"{category}_declarations": [],
-            },
-        )
+        if (
+            (whitelist_keys is not None)
+            and len(whitelist_keys) > 1
+            and (category == "NativeFunctions")
+        ):
+            # {name}_native.h is backend specific for whitelist mode
+            # here, we generate into {dispatch_key}/op/{name}_native.h separately.
+            for white_key in whitelist_keys:
+                dispatch_namespace = white_key.lower()
+                cpu_fm.write(
+                    f"{category}.h",
+                    lambda: {
+                        f"{category}_includes": [
+                            f"#include <ATen/{dispatch_namespace}/ops/{name}{suffix}.h>"
+                            for name in sorted(functions_by_root_name.keys())
+                        ],
+                        f"{category}_declarations": [],
+                    },
+                )
+        else:
```

Comment:
Do we need to handle multiple `white list keys` now? Should we just add an assert here? Like:
```python
assert not whitelist_keys or len(whiltelist_keys) == 1
```